llama-index-core/llama_index/core/evaluation/guideline.py

```diff
@@ -71,7 +79,9 @@ def __init__(
         else:
             self._eval_template = eval_template or DEFAULT_EVAL_TEMPLATE
 
-        self._output_parser = PydanticOutputParser(output_cls=EvaluationData)
+        self._output_parser = PydanticOutputParser(
+            output_cls=EvaluationData, pydantic_format_tmpl=PYDANTIC_FORMAT_TMPL
```

Comment:
Why do we modify prompt here? There is already a very similar prompt here
https://github.com/run-llama/llama_index/blob/3445b024bb583c8845190326b405608c38008c73/llama-index-core/llama_index/core/output_parsers/pydantic.py#L10

Feels like either the default prompt could be modified, or this is specific to the LLM you are using and you should just pass in a customized output parser as kwarg

Reply:
Yes. Actually I'm overwriting the template for evaluations only because I'm not quite sure whether it's a common requirements for json output to modify the default template. 
Now I feel adding a custom output parser to the constructor here makes more sense since it allows flexibility with different LLMs/settings and  backward compatibility. I have added a custom output parser para now.